### PR TITLE
[Common] New KVCacheMgr to support CB

### PR DESCRIPTION
**Pull request by pujiang2018**

### CODE REVIEW

**Comment by pujiang2018** [comment]
@Duyi-Wang @abenmao @changqi1 Pls take time to review this.

File: src/common/kvcache_mgr.h [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+#include "kvcache_tensor.h"
+#include <unordered_map>
+
+namespace xft {
+
+class KVCacheMgrImplBase {
+public:
+    virtual ~KVCacheMgrImplBase() = default;
+    virtual bool delSequence(int seqID) = 0;
+    virtual bool addSequence(int seqID, int prefixId = 0) = 0;
```
Comment (Duyi-Wang) [inline src/common/kvcache_mgr.h:27]: default value for prefixID is better to be -1? since the ID starts from 0.

File: src/common/kvcache_mgr.h [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+        std::vector<KVCacheTensor<T> *> readyList;
+        readyList.reserve(seqIDs.size());
+
+        for (auto seqID : seqIDs) {
+            auto it = sequenceCaches.find(seqID);
+            if (it == sequenceCaches.end()) { return false; }
+            readyList.push_back(it->second);
+        }
+
+        readyCaches == std::move(readyList);
```
Comment (Duyi-Wang) [inline src/common/kvcache_mgr.h:127]: =？

Comment (pujiang2018) [reply src/common/kvcache_mgr.h]: typo and fixed.

File: src/common/kvcache_mgr.h (hunk 1 above)
Comment (pujiang2018) [reply src/common/kvcache_mgr.h]: revised.

**Review by Duyi-Wang — APPROVED** [review]

**Pull request merged**